### ISSUE TITLE
[PPML] Occlum Spark TPC-H & unit test zlib zstd issue

### ISSUE THREAD

**Issue by qiyuangong** [intel-analytics/BigDL#4317]
SparkSQL Unit Test
```
java.lang.NoClassDefFoundError: Could not initialize class com.github.luben.zstd.ZstdInputStream
```

Spark TPC-H
```
22-03-23 01:58:56 [task-result-getter-1] WARN  TaskSetManager:69 - Lost task 2.0 in stage 1.0 (TID 2382) (10.244.0.239 executor 4): java.lang.NoClassDefFoun
dError: Could not initialize class com.github.luben.zstd.ZstdInputStream
        at org.apache.spark.io.ZStdCompressionCodec.compressedInputStream(CompressionCodec.scala:235)
        at org.apache.spark.MapOutputTracker$.deserializeObject$1(MapOutputTracker.scala:951)
        at org.apache.spark.MapOutputTracker$.deserializeMapStatuses(MapOutputTracker.scala:961)
        at org.apache.spark.MapOutputTrackerWorker.$anonfun$getStatuses$2(MapOutputTracker.scala:847)
        at org.apache.spark.util.KeyLock.withLock(KeyLock.scala:64)
        at org.apache.spark.MapOutputTrackerWorker.getStatuses(MapOutputTracker.scala:841)
        at org.apache.spark.MapOutputTrackerWorker.getMapSizesByExecutorId(MapOutputTracker.scala:815)
        at org.apache.spark.shuffle.sort.SortShuffleManager.getReader(SortShuffleManager.scala:133)
        at org.apache.spark.shuffle.ShuffleManager.getReader(ShuffleManager.scala:63)
        at org.apache.spark.shuffle.ShuffleManager.getReader$(ShuffleManager.scala:57)
        at org.apache.spark.shuffle.sort.SortShuffleManager.getReader(SortShuffleManager.scala:73)
        at org.apache.spark.sql.execution.ShuffledRowRDD.compute(ShuffledRowRDD.scala:190)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
        at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
        at org.apache.spark.scheduler.Task.run(Task.scala:131)
        at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1442)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

```
22-03-23 01:58:56 [task-result-getter-2] WARN  TaskSetManager:69 - Lost task 20.0 in stage 1.0 (TID 2400) (10.244.0.239 executor 4): java.lang.UnsatisfiedLi
nkError: /tmp/libzstd-jni1294625860834535936.so: /tmp/libzstd-jni1294625860834535936.so: failed to map segment from shared object
no zstd-jni in java.library.path: [/usr/lib/jvm/java-11-openjdk-amd64/lib/server, /usr/lib/jvm/java-11-openjdk-amd64/lib, /usr/lib/jvm/java-11-openjdk-amd64
/../lib, /lib, /usr/java/packages/lib, /usr/lib/x86_64-linux-gnu/jni, /lib/x86_64-linux-gnu, /usr/lib/x86_64-linux-gnu, /usr/lib/jni, /lib, /usr/lib]
Unsupported OS/arch, cannot find /linux/amd64/libzstd-jni.so or load zstd-jni from system libraries. Please try building from source the jar or providing li
bzstd-jni in your system.
        at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2670)
        at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:830)
        at java.base/java.lang.System.loadLibrary(System.java:1873)
        at com.github.luben.zstd.util.Native.load(Native.java:122)
        at com.github.luben.zstd.util.Native.load(Native.java:55)
        at com.github.luben.zstd.ZstdInputStream.<clinit>(ZstdInputStream.java:22)
        at org.apache.spark.io.ZStdCompressionCodec.compressedInputStream(CompressionCodec.scala:235)
        at org.apache.spark.MapOutputTracker$.deserializeObject$1(MapOutputTracker.scala:951)
        at org.apache.spark.MapOutputTracker$.deserializeMapStatuses(MapOutputTracker.scala:961)
        at org.apache.spark.MapOutputTrackerWorker.$anonfun$getStatuses$2(MapOutputTracker.scala:847)
        at org.apache.spark.util.KeyLock.withLock(KeyLock.scala:64)
        at org.apache.spark.MapOutputTrackerWorker.getStatuses(MapOutputTracker.scala:841)
        at org.apache.spark.MapOutputTrackerWorker.getMapSizesByExecutorId(MapOutputTracker.scala:815)
        at org.apache.spark.shuffle.sort.SortShuffleManager.getReader(SortShuffleManager.scala:133)
        at org.apache.spark.shuffle.ShuffleManager.getReader(ShuffleManager.scala:63)
        at org.apache.spark.shuffle.ShuffleManager.getReader$(ShuffleManager.scala:57)
        at org.apache.spark.shuffle.sort.SortShuffleManager.getReader(SortShuffleManager.scala:73)
        at org.apache.spark.sql.execution.ShuffledRowRDD.compute(ShuffledRowRDD.scala:190)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
        at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
        at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
        at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)

```

**Comment by qiyuangong** [comment]
Seems like previous occlum temp dir issue, which is fixed by occlum team in occlum 0.24. Will check with occlum team.

**Comment by qiyuangong** [comment]
Issue resolved by occlum image upgrade.
